packages/caravan-health/src/fingerprint.ts

```diff
@@ -0,0 +1,51 @@
+import { getAddressType, Network } from "@caravan/bitcoin";
+import { ScriptType } from "@caravan/fees";
+
+export function scriptTypeFingerprints(
+  inputScriptType: ScriptType,
+  outputAddresses: string[],
+  network: Network,
+): boolean[] {
+  return outputAddresses.map((address) => {
+    const addressType = getAddressType(address, network);
+    return addressType === inputScriptType;
+  });
+}
+
+function log2(x: number): number {
+  return Math.log(x) / Math.LN2;
+}
+
+function shannonEntropy(str: string): number {
+  const counts: Record<string, number> = {};
+  for (const char of str) {
+    counts[char] = (counts[char] || 0) + 1;
+  }
+  const total = str.length;
+  return Object.values(counts).reduce((entropy, count) => {
+    const probability = count / total;
+    return entropy - probability * log2(probability);
+  }, 0);
+}
+
+export function amountFingerprints(amounts: string[]): boolean[] {
```

Comment:
would you mind adding docstrings to this to explain what's being solved here and in particular to describe how to interpret the output.